### ISSUE TITLE
Add support for accessibility tests with Axe for React

### ISSUE THREAD

**Issue by kasya** [OWASP/Nest#3058]
Currently, our project does not have comprehensive accessibility tests for React components using Axe. This gap may lead to inaccessible features being introduced undetected.

Integrate Axe accessibility testing into the React workflow. Add automated tests to ensure React components meet accessibility standards, and set up CI checks so that accessibility issues are caught early in development.

Tests should be added to `frontend/__tests__/accessibility` folder and match our project structure.

**Are you going to work on implementing this?**

- [ ] Yes
- [x] No

**Additional context**
- Reference: https://www.deque.com/axe/

**Comment by Utkarsh-0304** [comment]
@kasya, can I work on this?

**Comment by anurag2787** [comment]
Hi @kasya I would like to give it a try could you please assign this to me?

**Comment by mrkeshav-05** [comment]
Hi @kasya , can I work on this?

**Comment by kasya** [comment]
@Utkarsh-0304 Hi! Assigned this to you! Please keep in mind the deadline. If that's not enough time for you since you might also need to fix any found issues - let me know before the deadline.

**Comment by SidharthxNST** [comment]
@kasya if thus issue is not solved pls assign this to me

**Comment by Utkarsh-0304** [comment]
> [@Utkarsh-0304](https://github.com/Utkarsh-0304) Hi! Assigned this to you! Please keep in mind the deadline. If that's not enough time for you since you might also need to fix any found issues - let me know before the deadline.

Hi @kasya, I just had one question. Do I need to test out the components only?

**Comment by kasya** [comment]
@Utkarsh-0304 I believe we'd want to test pages as well 🤔 I mean those in `frontend/src/app` directory. Many of those only have components inside them, but some also have some templating besides components.

**Comment by Utkarsh-0304** [comment]
> [@Utkarsh-0304](https://github.com/Utkarsh-0304) I believe we'd want to test pages as well 🤔 I mean those in `frontend/src/app` directory. Many of those only have components inside them, but some also have some templating besides components.

Sure! I have completed the accessibility tests for components. After this, I was looking into tests for pages, but there are just a lot of related issues that need to be fixed. I think this might not be completed within the deadline.

**Comment by arkid15r** [comment]
> > [@Utkarsh-0304](https://github.com/Utkarsh-0304) I believe we'd want to test pages as well 🤔 I mean those in `frontend/src/app` directory. Many of those only have components inside them, but some also have some templating besides components.
> 
> Sure! I have completed the accessibility tests for components. After this, I was looking into tests for pages, but there are just a lot of related issues that need to be fixed. I think this might not be completed within the deadline.

Feel free to split the issue to sub-issues as needed @Utkarsh-0304

**Comment by HarshitVerma109** [comment]
Hi @arkid15r , 
since the deadline for this task has passed, I'd like to work on it. Could you please assign it to me?

**Comment by udaykiran243** [comment]
Hey @Utkarsh-0304 if this issue wants any help I'm interested to step in and as @arkid15r mentioned we can split to sub-issues if needed.
Thanks!
